lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -0,0 +1,604 @@
+     * Parses passed value as JSON object, JSON.parse() will throw an error.
+     * @param input
+     */
+    private IsJSON(jsonValue: string): void {
+        JSON.parse(jsonValue);
+    }
+
+    /**
+     * fetches the entry from the browser storage based off the key
+     * @param key
+     */
+    getItem(key: string): string {
+        return this.browserStorage.getItem(key);
+    }
+
+    /**
+     * sets the entry in the browser storage
+     * @param key
+     * @param value
+     */
+    setItem(key: string, value: string): void {
+        this.browserStorage.setItem(key, value);
+    }
+
+    /**
+     * fetch the account entity from the platform cache
+     * @param accountKey
+     */
+    getAccount(accountKey: string): AccountEntity | null {
+        const account = this.getItem(accountKey);
+        if (StringUtils.isEmpty(account)) {
+            return null;
+        }
+
+        try {
+            this.IsJSON(account);
+            const accountEntity = CacheManager.toObject(new AccountEntity(), JSON.parse(account));
+            if (AccountEntity.isAccountEntity(accountEntity)) {
+                return accountEntity;
+            }
+            return null;
+        } catch (e) {
+            return null;
+        }
+    }
+
+    /**
+     * set account entity in the platform cache
+     * @param key
+     * @param value
+     */
+    setAccount(account: AccountEntity): void {
+        const key = account.generateAccountKey();
+        this.setItem(key, JSON.stringify(account));
+    }
+
+    /**
+     * generates idToken entity from a string
+     * @param idTokenKey
+     */
+    getIdTokenCredential(idTokenKey: string): IdTokenEntity {
+        const value = this.getItem(idTokenKey);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+        const idToken: IdTokenEntity = CacheManager.toObject(new IdTokenEntity(), JSON.parse(value));
+        return idToken;
+    }
+
+    /**
+     * set IdToken credential to the platform cache
+     * @param idToken
+     */
+    setIdTokenCredential(idToken: IdTokenEntity): void {
+        const idTokenKey = idToken.generateCredentialKey();
+        this.setItem(idTokenKey, JSON.stringify(idToken));
+    }
+
+    /**
+     * generates accessToken entity from a string
+     * @param key
+     */
+    getAccessTokenCredential(accessTokenKey: string): AccessTokenEntity {
+        const value = this.getItem(accessTokenKey);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+        const accessToken: AccessTokenEntity = CacheManager.toObject(new AccessTokenEntity(), JSON.parse(value));
+        return accessToken;
+    }
+
+    /**
+     * set accessToken credential to the platform cache
+     * @param accessToken
+     */
+    setAccessTokenCredential(accessToken: AccessTokenEntity): void {
+        const accessTokenKey = accessToken.generateCredentialKey();
+        this.setItem(accessTokenKey, JSON.stringify(accessToken));
```

Comment:
Can we please add validation for all cache entities?

Reply:
going to add in a follow up PR, don't want to make this PR larger than it has to be.

Reply:
Alright!